### ISSUE TITLE
Body parent and child Joint frames are indistinguishable

### ISSUE THREAD

**Issue by aseth1** [opensim-org/opensim-gui#550]
### Steps to reproduce
Load any model with at least one joint (e.g. arm26.osim)
Toggle display of Body and Joint frames (both parent and child).

### Expected result
Parent frame is bold/thicker to distinguish from the child frame. 
Red, green blue axes correspond to X, Y, Z of the frame.

### Actual result
All frames appear identical and are orange, florescent green and baby blue and very thin and difficult to see. 
![screen shot 2018-02-26 at 12 00 36 pm](https://user-images.githubusercontent.com/6494122/36692382-c2562788-1aec-11e8-95e6-65d59a673aec.png)

### Environment and GUI version
Mac 02-25

**Comment by aymanhab** [comment]
With components drawing themselves, Frame doesn't know which is which, API should change size/scale/etc. for these then visualizer should respect these.

**Comment by aymanhab** [comment]
Make thicker frames and the API will set thickness based on context, the visualizer will respect these.

**Comment by aseth1** [comment]
Exploring this further, changing thicknesses via the API is problematic. The main issue is that if we make thickness a property of `Frame` (or its geometry) and then have a `Joint` override it according to its context (parent or child) then a `Body` frame would potentially appear as a `Joint` frame for all time. Furthermore, a `Frame` could be both a child for one `Joint` and a parent to another. There is no way the API can know what display context is correct in that case. Only the GUI can make that call depending on what is being selected. 

When a user selects a Joint and clicks "Toggle Parent Frame" then the GUI knows the exact context. It seems in this case we need the frame geometry being displayed in the GUI to have attributes like size and/or thickness, cap type, ... that the GUI can modify (just like hiding or changing opacity) based on context. That is if I then select another Joint when the previous child body is now the parent and ask this Joint to toggle its frames I would see the Frame change display type from child to parent and see the new child frame emphasized as the child.

The issue seems to be that the built in frame geometry in three.js is just a fixed thing with no attributes we can use to make these contextual changes. One proposal then is to just create our own frame geometry composed of cylinders of different colors (e.g. RGB) and with the attributes we want to have control over without forcing the visualizer to populate new geometry.

Perhaps @tkuchida or others with an aesthetic eye can prototype a new frame geometry type with attributes that can be changed to make for nice looking parent and child frame representations.

**Comment by tkuchida** [comment]
I can't think of an unambiguous mapping from the set {"parent", "child"} to color, thickness, etc. Two ideas:
![](https://user-images.githubusercontent.com/4203505/37377867-b0c6d7e8-26e9-11e8-8d6b-05f5a55e25e3.png)
![](https://user-images.githubusercontent.com/4203505/37377870-b3fdedc0-26e9-11e8-86a2-7f0e7bf44d73.png)

**Comment by aseth1** [comment]
I love the first one! :+1: 

I suggested the second to @aymanhab but he said presenting text consistently in the 3D view could be challenging.

**Comment by tkuchida** [comment]
> presenting text consistently in the 3D view could be challenging

True. Should be possible, though. There's already an orthographic camera lying around for rendering the OpenSim logo, so some of the infrastructure (minus projecting from world coordinates into screen coordinates) is already available.

**Comment by jenhicks** [comment]
Make parent cube tipped and slightly bigger. Slightly smaller for the child and spheres
